system/java_certs.py

```diff
@@ -0,0 +1,218 @@
+
+rc:
+  description: Keytool command execution return value
+  returned: success
+  type: int
+  sample: "0"
+
+cmd:
+  description: Executed command to get action done
+  returned: success
+  type: string
+  sample: "keytool -importcert -noprompt -keystore"
+'''
+
+try:
+    import json
+except ImportError:
+    import simplejson as json
+
+import os
+import subprocess
+import sys
+import datetime
+import syslog
+
+def check_cert_present(module, keystore_path, keystore_pass, alias):
+    test_cmd = "keytool -noprompt -list -keystore '%s' -storepass '%s' -alias '%s'" % (keystore_path, keystore_pass, alias)
+
+    (rc, out, err) = module.run_command(test_cmd)
+    if rc == 0:
+        return True
+    return False
+
+def import_cert_url(module, url, port, keystore_path, keystore_pass, alias):
+    fetch_cmd = "keytool -printcert -rfc -sslserver %s:%s" % (url, port)
+    import_cmd = "keytool -importcert -noprompt -keystore '%s' -storepass '%s' -alias '%s'" % (keystore_path, keystore_pass, alias)
+
+    # Fetch SSL certificate from remote host.
+    (rc, fetch_out, fetch_err) = module.run_command(fetch_cmd, check_rc=True)
+
+    # Use remote certificate from remote host and import it to a java keystore
+    (rc, import_out, import_err) = module.run_command(import_cmd, data=fetch_out, check_rc=False)
+    if rc == 0:
+        return module.exit_json(changed=True, msg=import_out,
+            rc=rc, cmd=import_cmd, stdout_lines=import_out)
+    else:
+        return module.fail_json(msg=import_out, rc=rc, cmd=import_cmd)
+
+
+def import_cert_path(module, path, keystore_path, keystore_pass):
+    import_cmd = "keytool -importcert -noprompt -keystore '%s' -storepass '%s' -file '%s'" % (keystore_path, keystore_pass, path)
+
+    # Use local certificate from local path and import it to a java keystore
+    (rc, import_out, import_err) = module.run_command(import_cmd, check_rc=False)
+    if rc == 0:
+        return module.exit_json(changed=True, msg=import_out,
+            rc=rc, cmd=import_cmd, stdout_lines=import_out)
+    else:
+        return module.fail_json(msg=import_out, rc=rc, cmd=import_cmd)
+
+
+def delete_cert(module, keystore_path, keystore_pass, alias):
+    del_cmd = "keytool -delete -keystore '%s' -storepass '%s' -alias '%s'" % (keystore_path, keystore_pass, alias)
+
+    # Delete SSL certificate from keystore
+    (rc, del_out, del_err) = module.run_command(del_cmd, check_rc=True)
+
+    return module.exit_json(changed=True, msg=del_out,
+        rc=rc, cmd=del_cmd, stdout_lines=del_out)
+
+def test_keytool(module):
+    ''' Test if keytool is actuall executable or not '''
+    test_cmd = "keytool"
+
+    (rc, del_out, del_err) = module.run_command(test_cmd, check_rc=True)
+
+def test_keystore(module, keystore_path):
+    ''' Check if we can access keystore as file or not '''
+    if keystore_path is None:
+        keystore_path=''
+
+    if not os.path.exists(keystore_path) and not os.path.isfile(keystore_path):
```

Comment:
Why do we fail if it doesn't exist? I disabled the test and keytool ran just fine with creating a new keystore, which was what I wanted. Maybe I'm missing something here...